File: Python/random_number.py
import sys
import random

def main():
    if len(sys.argv) != 2:
        print("Usage: python script.py <n>")
        return
    
    try:
        n = int(sys.argv[1])
        if n < 1 or n > 100:
            raise ValueError
    except ValueError:
        print("Please provide a positive integer between 1 and 100.")
        return
    
    random_numbers = [random.random() for _ in range(n)]
    
    with open("random.txt", "w") as f:
        for num in random_numbers:
            f.write(f"{num}\n")
    
    print(f"生成的随机数已保存到 random.txt 文件.")
    
    with open("random.txt", "r") as f:
        random_numbers_from_file = [float(line.strip()) for line in f.readlines()]
    
    print(f"从 random.txt 文件中读取的随机数: {random_numbers_from_file}")
    
    sorted_numbers = sorted(random_numbers_from_file)
    
    with open("sorted.txt", "w") as f:
        for num in sorted_numbers:
            f.write(f"{num}\n")
    
    print(f"排序后的随机数已保存到 sorted.txt 文件.")

if __name__ == "__main__":
    main()

File: Rust/src/bin/random_number.rs
use rand::Rng;
use std::env;
use std::fs::File;
use std::io::{self, BufRead, Write, BufReader};

fn main() -> io::Result<()> {
    let args: Vec<String> = env::args().collect();

    if args.len() != 2 {
        eprintln!("Usage: {} <n>", args[0]);
        std::process::exit(1);
    }

    let n: usize = match args[1].parse() {
        Ok(num) if num >= 1 && num <= 100 => num,
        _ => {
            eprintln!("Please provide a valid number between 1 and 100.");
            std::process::exit(1);
        }
    };

    let mut rng = rand::thread_rng();
    let random_numbers: Vec<f64> = (0..n).map(|_| rng.gen_range(0.0..=1.0)).collect();

    let mut random_file = File::create("random.txt")?;
    for num in &random_numbers {
        writeln!(random_file, "{}", num)?;
    }
    println!("Random numbers saved to random.txt");

    let file = File::open("random.txt")?;
    let reader = BufReader::new(file);
    let mut loaded_numbers: Vec<f64> = Vec::new();
    for line in reader.lines() {
        let num: f64 = line?.parse().unwrap();
        loaded_numbers.push(num);
    }

    loaded_numbers.sort_by(|a, b| a.partial_cmp(b).unwrap());

    let mut sorted_file = File::create("sorted.txt")?;
    for num in &loaded_numbers {
        writeln!(sorted_file, "{}", num)?;
    }
    println!("Sorted numbers saved to sorted.txt");

    Ok(())
}
